### PR TITLE
[BREAKING] Remove non-0.5.0 warning for emit keyword (make it mandatory)

### PR DESCRIPTION
**Pull request by axic**
Pulled off #4179.

### CODE REVIEW

File: test/libsolidity/syntaxTests/types/empty_tuple_event.sol [hunk 1]
```diff
@@ -2,9 +2,9 @@ pragma solidity ^0.4.3;
 contract C {
     event SomeEvent();
     function a() public {
-        (SomeEvent(), 7);
+        (emit SomeEvent(), 7);
```
Comment (axic) [inline test/libsolidity/syntaxTests/types/empty_tuple_event.sol:5]: This with `emit SomeEvent` is not parsed correctly - "expected primary expression". Either we can just remove the test case or "fix" the parser.

Comment (chriseth) [reply test/libsolidity/syntaxTests/types/empty_tuple_event.sol:5]: We remove the test case or require it to fail. I specifically wrote the parser to only accept `emit` at statement level.

Comment (chriseth) [reply test/libsolidity/syntaxTests/types/empty_tuple_event.sol:5]: Or was that only changed later? Anyway, EmitStatements should be statements :)

**Comment by chriseth** [comment]
Fine apart from failing tests.

**Comment by chriseth** [comment]
(Hopefully) fixed the tests.

File: test/libsolidity/SolidityEndToEndTest.cpp [hunk 2]
```diff
@@ -3186,28 +3186,6 @@ BOOST_AUTO_TEST_CASE(event_no_arguments)
 	BOOST_CHECK_EQUAL(m_logs[0].topics[0], dev::keccak256(string("Deposit()")));
 }
 
-BOOST_AUTO_TEST_CASE(event_access_through_base_name)
```
Comment (chriseth) [inline test/libsolidity/SolidityEndToEndTest.cpp:3189]: now duplicate

File: test/libsolidity/SolidityEndToEndTest.cpp [hunk 3]
```diff
@@ -3238,67 +3216,15 @@ BOOST_AUTO_TEST_CASE(events_with_same_name)
 			event Deposit(address _addr);
 			event Deposit(address _addr, uint _amount);
 			function deposit() returns (uint) {
-				Deposit();
```
Comment (chriseth) [inline test/libsolidity/SolidityEndToEndTest.cpp:3241]: now duplicate

**Review by chriseth — APPROVED** [review]

File: test/libsolidity/SolidityEndToEndTest.cpp (hunk 2 above)
Comment (axic) [reply test/libsolidity/SolidityEndToEndTest.cpp:3189]: Which one is this a duplicate of?

Comment (axic) [reply test/libsolidity/SolidityEndToEndTest.cpp:3189]: I see the one below.

File: test/libsolidity/SolidityEndToEndTest.cpp (hunk 3 above)
Comment (axic) [reply test/libsolidity/SolidityEndToEndTest.cpp:3241]: Where is this duplicated?

Comment (chriseth) [reply test/libsolidity/SolidityEndToEndTest.cpp:3241]: `events_with_same_name_inherited` and `events_with_same_name_inherited_emit` are duplicated.

File: test/libsolidity/SolidityEndToEndTest.cpp [hunk 4]
```diff
@@ -3238,67 +3216,15 @@ BOOST_AUTO_TEST_CASE(events_with_same_name)
...
-
-	ABI_CHECK(callContractFunction("deposit(address,uint256)", c_loggedAddress, u256(100)), encodeArgs(u256(1)));
-	BOOST_REQUIRE_EQUAL(m_logs.size(), 1);
-	BOOST_CHECK_EQUAL(m_logs[0].address, m_contractAddress);
-	BOOST_CHECK(m_logs[0].data == encodeArgs(c_loggedAddress, 100));
-	BOOST_REQUIRE_EQUAL(m_logs[0].topics.size(), 1);
-	BOOST_CHECK_EQUAL(m_logs[0].topics[0], dev::keccak256(string("Deposit(address,uint256)")));
-}
-
-BOOST_AUTO_TEST_CASE(events_with_same_name_inherited)
```
Comment (chriseth) [inline test/libsolidity/SolidityEndToEndTest.cpp:3279]: The above comment should have been here. The diff is a little weird, I deleted this test and did not touch the one above.

File: test/libsolidity/SolidityEndToEndTest.cpp (hunk 3 above)
Comment (chriseth) [reply test/libsolidity/SolidityEndToEndTest.cpp:3241]: See below.

**Pull request merged**